webui/src/Updates/UpdateLoginStatus.elm

```diff
@@ -65,20 +65,25 @@ updateLoginStatus message model =
                 )
 
         FetchVerify (Err error) ->
-            case error of
-                BadStatus response ->
-                    case response.status.code of
-                        403 ->
-                            ( { model | authRequired = Just True }
-                            , CmdUtils.delayMsg (5 * Time.second) AttemptReconnect
-                            )
+            if (not (model.wsConnected)) then
+                case error of
+                    BadStatus response ->
+                        case response.status.code of
+                            403 ->
+                                ( { model | authRequired = Just True }
+                                , CmdUtils.delayMsg (5 * Time.second) AttemptReconnect
+                                )
 
-                        _ ->
-                            ( model
-                            , CmdUtils.delayMsg (5 * Time.second) AttemptReconnect
-                            )
+                            _ ->
+                                ( model
+                                , CmdUtils.delayMsg (5 * Time.second) AttemptReconnect
+                                )
 
-                _ ->
-                    ( model
-                    , CmdUtils.delayMsg (5 * Time.second) AttemptReconnect
-                    )
+                    _ ->
```

Comment:
Is it possible to combine type checking of the `response` and the value comparision of its attribute `status.code`? This would avoid code duplication and make this snippet a bit more readable.

Reply:
I see your point. The problem is that I first need to deconstruct the BadStatus before I can get the response. I tried pattern matching the status code directly but it didn't work and I couldn't find any information online.

In Scala you can use [guards](https://alvinalexander.com/scala/how-to-use-if-then-expressions-guards-in-case-statements-scala) for this but Elm doesn't have this feature.

What I will do is at least factor our the reconnect command `CmdUtils.delayMsg (5 * Time.second) AttemptReconnect` into a separate function and use it everywhere.

Does that work for you?

Reply:
Yup, I also tried to find a way to pattern match the type and the attribute value, but did not succeed.

Sure, what you suggested makes sense.

Thank you!

Reply:
Done!